docs/guides/custom-functions.md

```diff
@@ -233,6 +233,18 @@ module.exports = (targetVal, _opts, paths) => {
 };
 ```
 
+Furthermore, if your function performs any fs calls to obtain dictionaries and similar, you may want to leverage cache.
+Each custom function is provided with its **own** cache instance that has a function-live lifespan.
+What does "function-live lifespan" mean? It means the cache is persisted for the whole life of a particular function.
+The cache will be retained between subsequent function calls and is never invalidated unless you compile the function again, i.e. load a ruleset again.
+In other words:
+- if you Spectral programmatically via JS API and your ruleset remains unchanged, all subsequent `spectral.run` calls will invoke custom functions with the same cache instance.
+As soon as you set a ruleset using `setRuleset` or `loadRuleset` method, each custom function will receive a new cache instance.
+- if you are a CLI user, the cache will never be invalidated.
+
+Please bear in mind that while you are welcome to store certain kind of data, using cache for exchanging information between subsequent function calls it strongly discouraged.
```

Comment:
@P0lip From a security standpoint, it's also highly discouraged to store sensitive information in this cache.